test/include/points.hpp

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include <random>
+
+#include <RGLFields.hpp>
+#include <math/Mat3x4f.hpp>
+#include <testing.hpp>
+
+static std::random_device randomDevice;
+static auto randomSeed = randomDevice();
+static std::mt19937 randomGenerator{randomSeed};
+
+class PointCloud
+{
+public:
+	explicit PointCloud(const std::vector<rgl_field_t>& declaredFields, std::size_t pointCount) : fields(declaredFields)
+	{
+		resize(pointCount);
+
+		std::size_t offset = 0;
+		for (const auto& field : fields) {
+			offsets[field] = offset;
+			offset += getFieldSize(field);
+		}
+	}
+
+	static PointCloud createFromNode(rgl_node_t outputNode, const std::vector<rgl_field_t>& fields)
+	{
+		int32_t outCount;
+		EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(outputNode, fields.at(0), &outCount, nullptr));
+
+		PointCloud pointCloud = PointCloud(fields, outCount);
+
+		for (auto& field : fields) {
+			int32_t currentCount, currentSize;
+			EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(outputNode, field, &currentCount, &currentSize));
+			EXPECT_EQ(currentSize, getFieldSize(field));
+			EXPECT_EQ(outCount, currentCount);
+
+			if (isDummy(field)) {
+				continue;
+			}
+
+			std::vector<char> dataFromNode;
+			dataFromNode.resize(currentCount * currentSize);
+			EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(outputNode, field, dataFromNode.data()));
+			for (int j = 0; j < currentCount; j++) {
+				std::move(dataFromNode.begin() + j * currentSize, dataFromNode.begin() + (j + 1) * currentSize,
+				          pointCloud.data.data() + j * pointCloud.getPointByteSize() + pointCloud.offsets.at(field));
+			}
+		}
+
+		return pointCloud;
+	}
+
+	std::size_t getPointCount() const { return data.size() / getPointByteSize(); }
+
+	void transform(const Mat3x4f& transform)
+	{
+		if (std::find(fields.begin(), fields.end(), XYZ_F32) == fields.end()) {
+			throw std::invalid_argument("PointCloud::transform: PointCloud does not contain XYZ_F32 field");
```

Comment:
In the near future, we have to separate XYZ from distance+vector spaces. In that case, this condition will not be true anymore.
I would leave a small comment with todo here about it.